src/connectors/impls/gpubsub/gsub.rs

```diff
@@ -0,0 +1,331 @@
+// Copyright 2022, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::connectors::google::AuthInterceptor;
+use crate::connectors::prelude::*;
+use crate::connectors::utils::url::HttpsDefaults;
+use async_std::channel::{Receiver, Sender};
+use async_std::stream::StreamExt;
+use async_std::task::JoinHandle;
+use dashmap::DashMap;
+use googapis::google::pubsub::v1::subscriber_client::SubscriberClient;
+use googapis::google::pubsub::v1::{PubsubMessage, ReceivedMessage, StreamingPullRequest};
+use gouth::Token;
+use serde::Deserialize;
+use std::fmt::Debug;
+use std::sync::Arc;
+use std::time::Duration;
+use tonic::codegen::InterceptedService;
+use tonic::transport::{Certificate, Channel, ClientTlsConfig};
+use tonic::Status;
+use tremor_pipeline::ConfigImpl;
+
+#[derive(Deserialize, Clone)]
+struct Config {
+    pub connect_timeout: u64,
+    pub ack_deadline: u64,
+    pub subscription_id: String,
+    #[serde(default = "default_endpoint")]
+    pub endpoint: String,
+    #[serde(default = "default_skip_authentication")]
```

Comment:
Is this used outside of tests? If not can we switch this to a #[cfg(test)] or something?